x/interchainquery/keeper/msg_server.go

```diff
@@ -150,19 +179,7 @@ func (k msgServer) SubmitQueryResponse(goCtx context.Context, msg *types.MsgSubm
 		return nil, err
 	}
 
-	// Verify the query hasn't expired (if the block time is greater than the TTL timestamp, the query is expired)
-	currBlockTime, err := cast.ToUint64E(ctx.BlockTime().UnixNano())
-	if err != nil {
-		return nil, err
-	}
-	if query.TimeoutTimestamp < currBlockTime {
-		k.Logger(ctx).Error(utils.LogICQCallbackWithHostZone(query.ChainId, query.CallbackId,
-			"QUERY TIMEOUT - QueryId: %s, TTL: %d, BlockTime: %d", query.Id, query.TimeoutTimestamp, ctx.BlockHeader().Time.UnixNano()))
-		return &types.MsgSubmitQueryResponseResponse{}, nil
-	}
-
 	// Immediately delete the query so it cannot process again
-	// We don't want to delete timed-out queries because they'll be retried
 	k.DeleteQuery(ctx, query.Id)
```

Comment:
Moving the timeout check below this line changes the bevahior: now all queries will be deleted even if timed out. I think this is desired, but want to double check with you that that's intentional + newly introduced behavior.

Reply:
yup that's right and it's how it's done on mainnet right now

An earlier commit to LSM changed to behavior to _not_ delete the query so that it can be scooped & retried in the end blocker. But that change is essentially reverted by this PR